pydarn/exceptions/borealis_exceptions.py

```diff
@@ -0,0 +1,353 @@
+                                 allowed_types=allowed_types)
+        Exception.__init__(self, self.message)
+
+
+class BorealisConvert2IqdatError(Exception):
+    """
+    Raised when the file cannot be converted to DARN DMap iqdat format.
+
+    Parameters
+    ----------
+    error_str: str
+        explanation for why the file cannot be converted to DARN iqdat.
+
+    Attributes
+    ----------
+    message: str
+        The message to display with the error
+    """
+
+    def __init__(self, error_str: str):
+        self.message = "The file cannot be converted to DARN iqdat due to "\
+            "the following error: {error_str}"\
+            "".format(error_str=error_str)
+        Exception.__init__(self, self.message)
+
+
+class BorealisConvert2RawacfError(Exception):
+    """
+    Raised when the file cannot be converted to DARN DMap rawacf format.
+
+    Parameters
+    ----------
+    error_str: str
+        explanation for why the file cannot be converted to DARN rawacf.
+    
+    Attributes
+    ----------
+    message: str
+        The message to display with the error
+    """
+
+    def __init__(self, error_str: str):
+        self.message = "The file cannot be converted to DARN rawacf due to "\
+            "the following error: {error_str}"\
+            "".format(error_str=error_str)
+        Exception.__init__(self, self.message)
+
+
+class BorealisRestructureError(Exception):
+    """
+    Raised when the file cannot be restructured to or from site/array 
+    styles. 
+
+    Parameters
+    ----------
+    error_str: str
+        explanation for why the file cannot be restructured.
+    
+    Attributes
+    ----------
+    message: str
+        The message to display with the error
+
+    See Also
+    --------
+    restructure_borealis.py
+    """
+
+    def __init__(self, error_str: str):
+        self.message = "The file cannot be restructured due to the "\
+            " following error: {error_str}"\
+            "".format(error_str=error_str)
+        Exception.__init__(self, self.message)
+
+
+class BorealisStructureError(Exception):
+    """
+    Raised when the file has major structural issues and may not
+    be the correct style type (arrays vs records).
+
+    Parameters
+    ----------
+    error_str: str
+        explanation for why the file cannot be restructured.
+    
+    Attributes
+    ----------
+    message: str
+        The message to display with the error
+
+    See Also
+    --------
+    restructure_borealis.py
+    """
+
+    def __init__(self, error_str: str):
+        self.message = "The file has major structural errors. You may be"\
+            " attempting to use the file as the wrong style (site vs. array style)."\
+            "Please check if you are interested in records or arrays. {error_str}"\
+            "".format(error_str=error_str)
+        Exception.__init__(self, self.message)
+
+
+class ConvertFileOverWriteError(Exception):
```

Comment:
I used this for routines that convert files (restructuring, SDARN DMap conversion, etc.). The input filename should not be the same as the output filename. I think this error will be more clear than an HDF5 error

Reply:
Why can't the output filename be the same as input filename? 
I noticed you use a temporary file for writing so it seems plausible to have the same input and output filename. Again, it may be something a user wants or not. we can leave it for now and see what people think.